pkg/tfshim/sdk-v2/provider.go

```diff
@@ -62,9 +63,21 @@ func NewProvider(p *schema.Provider, opts ...providerOption) shim.Provider {
 		tf:   p,
 		opts: opts,
 	}
-	if cmdutil.IsTruthy(os.Getenv("PULUMI_ENABLE_PLAN_RESOURCE_CHANGE")) {
+
+	const envPRCVar = "PULUMI_ENABLE_PLAN_RESOURCE_CHANGE"
+	envPRC := os.Getenv(envPRCVar)
+	switch {
+	case cmdutil.IsTruthy(envPRC):
 		return newProviderWithPlanResourceChange(p, prov, func(s string) bool { return true })
+	case strings.EqualFold(envPRC, "false") || envPRC == "0":
+		return prov
+	case "":
+		// Do nothing, we will apply the default below.
+	default:
+		// The user set PULUMI_ENABLE_PLAN_RESOURCE_CHANGE to a value that isn't valid.
+		warn("%s is not a valid value for %s", envPRC, envPRCVar)
```

Comment:
This is not a function. I wasn't sure what the correct "warn" mechanism was here, so I left `warn` as a placeholder for "warn the user".

Reply:
I don't know either so let's merge as is as this is low-value either way.

This is a debug only env var, primarily to be used by us. If any user is using it, it will be after we've given them the command, so should be ok.

Also BLAH=disable likely doesn't work anywhere.